### PR TITLE
Add lightning-latest-rc.yml

### PR DESCRIPTION
**Pull request by vincentmr**
`lightning-latest-rc.yml` was removed and should be added back.

### CODE REVIEW

**Comment by vincentmr** [comment]
Restore testing against PennyLane RC in plugin [sc-48584]

**Review by mudit2812 — COMMENTED** [review]
Thanks Vincent! Could you also add this test to the `compile.py` file. We use that script so that we can simply update a workflow template and then use that script to update all workflows automatically rather updating all workflows manually.

**Comment by timmysilv** [comment]
on every release cycle, someone changes the RC version in the template then runs `python compile.py` to bump the RC version in each workflow. because lightning isn't in compile.py, it won't be auto-updated and it will most likely be forgotten. I'm not sure the best way to fix this, because this does seem to be a bit different from other plugins. basically we should just ensure that compile.py bumps the RC version in this file when it's been changed in the template, or maybe something else?

**Comment by vincentmr** [comment]
After trying it out, I see that `compile.py` does not produce the correct `lightning-latest-rc.yml` file, which is a little special. It will be more so when we add L-GPU which will have to be tested on a self-hosted GPU runner.

**Comment by vincentmr** [comment]
So should I revert `compile.py`?

**Comment by mudit2812** [comment]
> So should I revert compile.py?

I think that's fine. I can just add a note to the release guide for future release managers to update the RC tests for lightning separately. Would be nice to have this resolved eventually though. @timmysilv thoughts?

**Comment by timmysilv** [comment]
I suppose it's fine for now, but I'd rather something more automatic. @vincentmr where did you get that template from? was it being auto-generated in the past?

**Comment by vincentmr** [comment]
I did not use a template. If we only want to test against lightning-qubit I think the template will do. Otherwise, I'm not sure how to pull and build lightning-kokkos, say, in the way the template does it because we need to install both lightning-qubit and lightning-kokkos (passing some env var to `pip`).

File: compile.py [hunk 1]
```diff
@@ -132,6 +132,13 @@
         "tests_loc": "tests/unit_test",
         "token": "QI_TOKEN",
     },
+    {
+        "plugin": "lightning",
+        "gh_user": "PennyLaneAI",
+        "which": [],
+        "requirements": [],
+        "device_tests": [],
+    },
```
Comment (timmysilv) [inline compile.py:141]:
```suggestion
```

**Review by timmysilv — APPROVED** [review]
don't worry about it then, would rather get this in. We'll update our docs for now.

please take my change removing lightning from compile.py before merging - you can leave the other one, definitely an unnecessary loop!

**Review by mudit2812 — APPROVED** [review]

**Pull request merged**